status/status.go

```diff
@@ -0,0 +1,91 @@
+package status
+
+import "sync"
+
+// Status represents a status update from an action within Empire.
+type Status interface {
+	// Returns a human readable string about the status update.
+	String() string
+}
+
+// status implements the Status interface
+type status struct {
+	message string
+}
+
+func (s *status) String() string {
+	return s.message
+}
+
+// NewStatus returns the default implementation of the Status interface
+func NewStatus(m string) *status {
+	return &status{message: m}
+}
+
+// StatusStream is an interface for publishing status updates while executing
+// an Empire action.
+type StatusStream interface {
+	// Publish publishes an update to the status stream
+	Publish(Status)
+
+	// Done finalizes the status stream
+	Done(error)
+}
+
+type SubscribableStream interface {
```

Comment:
i like that.

another thing i'm coming across is having to do: https://github.com/remind101/empire/pull/917/files#diff-48d76aef8c283046a79db32824933290R523.

not the end of the world, but definitely annoying.

i was thinking of adding an `active` flag within the stream, if it never gets activated, then the subscriber never blocks. there could either be a specific `activate` method or as soon as you `publish` the stream would get activated. but that defeats the purpose of being able to write to the stream at any point within a go routine and having the client subscribe.

also, i'm not a fan of throwing: `panic("Publish called on finalized stream")`. given these are just status updates, it feels aggressive to throw a panic. what would be the proper way to warn instead? i couldn't find any examples in the code.